ingestion/functions/parsing/japan/japan.py

```diff
@@ -43,23 +40,48 @@ def convert_location(raw_entry):
         if term and term != "Unspecified"]
     return {"query":  ", ".join(query_terms)}
 
-
 def detect_notes(raw_notes):
     if "notes" in raw_notes:
         return raw_notes["notes"]
     else:
         return None
 
-
 def convert_date(raw_date: str):
     """
     Convert raw date field into a value interpretable by the dataserver.
-
+​
     The date filtering API expects mm/dd/YYYYZ format.
     """
-    date = datetime.strptime(raw_date, "%Y-%m-%d")
+    date = datetime.strptime(raw_date["dateAnnounced"], "%Y-%m-%d")
     return date.strftime("%m/%d/%YZ")
 
+def convert_additionalSources(additional_sourceURL: str):
```

Comment:
Isn't this `Dict`, rather than `str`? Likewise for `convert_outcome` and `convert_date`.

https://docs.python.org/3/library/typing.html

Reply:
sorry yes, copy pasted some functions and forgot to change it... Thanks!

Reply:
Isn't the input a str here?

Reply:
No it's a dict